target/scripts/helpers/utils.sh

```diff
@@ -105,15 +122,16 @@ function _replace_by_env_in_file() {
   fi
 
   local ENV_PREFIX=${1} CONFIG_FILE=${2}
+  local KV_DELIMITER=${3:-'='}
   local ESCAPED_VALUE ESCAPED_KEY
 
-  while IFS='=' read -r KEY VALUE; do
+  while IFS="${KV_DELIMITER}" read -r KEY VALUE; do
     KEY=${KEY#"${ENV_PREFIX}"} # strip prefix
-    ESCAPED_KEY=$(sed -E 's#([\=\&\|\$\.\*\/\[\\^]|\])#\\\1#g' <<< "${KEY,,}")
-    ESCAPED_VALUE=$(sed -E 's#([\=\&\|\$\.\*\/\[\\^]|\])#\\\1#g' <<< "${VALUE}")
+    ESCAPED_KEY=$(_escape_for_sed "${KEY,,}")
```

Comment:
Why is `$KEY` lower-cased? Currently this function is only used for LDAP. To keep it universal (able to handle other files which may have mixed cases), `$KEY` shouldn't be altered.

Reply:
> To keep it universal (able to handle other files which may have mixed cases), `$KEY` shouldn't be altered.

While I could agree with you there. I've not changed the existing behaviour, this was how the variable was handled previously.

It's only used for LDAP presently, but once my Config Templates PR is merged, this helper has no need to be tied to LDAP configs as it won't be used by anything.

I would ask if the lowercasing was dropped, how would it handle uppercase ENV keys matching lowercase config keys? It is probably better to defer that sort of change until a feature requires it. We'd then have an actual known use-case for it.

Presently it's an internal helper, so nothing but our scripts should be using it. I'm just providing the revision here as an extracted follow-up from the Config Templates PR. The functionality can remain without being used, but if it's dropped and later is brought back at least it'll be this improved/fixed version 😅